### ISSUE TITLE
HttpProtocol Metadata Options and Tags Not Supported in All Regions

### ISSUE THREAD

**Issue by cvockrodt** [hashicorp/terraform-provider-aws#25227]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.15.3
on linux_amd64
+ provider hashicorp/aws v4.17.1

### Affected Resource(s)

* aws_launch_template

### Terraform Configuration Files

```hcl
data "aws_availability_zones" "available" {
  state = "available"
}

resource "aws_launch_template" "test" {
  name "test"
  metadata_options {
    http_endpoint = "enabled"
  }
}

resource "aws_autoscaling_group" "test" {
  availability_zones = data.aws_availability_zones.available.names
  desired_capacity = 1
  min_size = 1
  max_size = 1

  launch_template {
    id = aws_launch_template.test.id
    version = aws_launch_template.test.latest_version
  }
}
```

### Debug Output

```xml
400 Bad Request
<Error>
  <Type>Sender</Type>
  <Code>ValidationError</Code>
  <Message>You must use a valid fully-formed launch template. Specifying HttpProtocol metadata options for an instance is not supported in this region.</Message>
</Error>
```
### Expected Behavior

Autoscaling Groups (ASG, EKS Node Group, etc.) should create successfully in all regions when associated with a launch template that specifies metadata_options

### Actual Behavior

Instances fail to create, with the following terraform error

```
Error: creating Auto Scaling Group (terraform-20223546789): ValidationError: You must use a valid fully-formed launch template. Specifying HttpProtocol metadata options for an instance is not supported in this region.
           status code: 400
    with aws_autoscaling_group.test
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Region in use is `us-iso-east-1`.

**Comment by justinretzolk** [comment]
Hey @cvockrodt 👋 Thank you for taking the time to raise this! To clarify a bit, is your request here that the launch template with `metadata_options` specified should work in all regions, or that Terraform should handle the error differently? The error you received is coming from the AWS API response, so unfortunately is not something Terraform can get around.

**Comment by cvockrodt** [comment]
Yeah I apologize, a little unclear.

The launch template should not apply default values for metadata options in regions that don't support launching instances with those options. The AWS API is falling short here by not dropping unsupported options from the launch template, but their response to me is just "update your launch templates to be compliant with the us-iso-east-1 region prior to executing them". In lieu of any AWS API changes, there are a few options. We can
1. update the AWS terraform provider to not send default values for certain regions (preferred?)
2. update the AWS terraform provider to not send default values at all
3. use an older version of the AWS terraform provider (< 3.59.0)
4. something else?

For option 1, I'm not sure if there's a way to determine via API which regions support which metadata options, so it may need to be a hard coded list :frowning:

**Comment by justinretzolk** [comment]
Hey @cvockrodt -- thank you for that additional information! You mentioning `< 3.59.0` really helped; I see that's when `http_protocol_ipv6` support was added. Interestingly, my read is that the provider is already [not sending a default value for `http_protocol_ipv6`](https://github.com/hashicorp/terraform-provider-aws/blob/06ebaf7182bf783e334d3c479d4efe2dd8377401/internal/service/ec2/ec2_launch_template.go#L1887-L1889), but I might be missing a key factor here. I'm going to mark this as a bug so that someone on the team can take a look at it as soon as time allows.

**Comment by cvockrodt** [comment]
Thanks!

That code certainly checks to see if the value is empty before adding it to the API request object, but the argument in the launch template resource does have a default value [according to the docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#http_protocol_ipv6), so I don't think the value will ever be empty at that conditional. You can see the default in the [schema definition](https://github.com/hashicorp/terraform-provider-aws/blob/06ebaf7182bf783e334d3c479d4efe2dd8377401/internal/service/ec2/ec2_launch_template.go#L646) in the same file as well.

Note: The behavior is a little odd because if you don't specify `metadata_options` at all, then no value is sent for `http_protocol_ipv6`, but if you specify anything in `metadata_options`, then all arguments are sent with their default values

**Comment by cvockrodt** [comment]
bump

**Comment by Simple-Analysis** [comment]
Also experiencing the same issue. This is especially relevant when we have to update the metadata options manually, after the instances have been provisioned. Then any subsequent terraform updates seem to error because it attempts to update the metadata options. 

Any luck in finding a good workaround?

**Comment by cvockrodt** [comment]
There is no workaround other than not using terraform to manage your node groups. I really hope this gets some attention.

**Comment by justinretzolk** [comment]
Hey all 👋 Thank you very much for taking the time to raise this! This was addressed with #30545, which was included in version `5.0.0` of the provider. With that in mind, we'll close this issue. If you experience additional issues with the provider, please do open a new issue to let us know.
